sapi/cli/php_cli_server.c

```diff
@@ -1680,14 +1680,37 @@ static void php_cli_server_client_save_header(php_cli_server_client *client)
 {
 	/* Wrap header value in a zval to add is to the HashTable which acts as an array */
 	zval tmp;
-	ZVAL_STR(&tmp, client->current_header_value);
 	/* strip off the colon */
 	zend_string *lc_header_name = zend_string_tolower_ex(client->current_header_name, /* persistent */ true);
 	GC_MAKE_PERSISTENT_LOCAL(lc_header_name);
 
-	/* Add the wrapped zend_string to the HashTable */
-	zend_hash_add(&client->request.headers, lc_header_name, &tmp);
-	zend_hash_add(&client->request.headers_original_case, client->current_header_name, &tmp);
+	zval *entry = zend_hash_find(&client->request.headers, lc_header_name);
+	bool is_forwarded_header = strstr(ZSTR_VAL(lc_header_name), "forwarded");
+
+	/**
+	 * **Forwarded** HTTP family headers can have 1 or more values separated by a comma while still
+	 * possibly be set separately by the client.
```

Comment:
This is not unique about the forwarded family. Per [RFC 9110#5.3](https://httpwg.org/specs/rfc9110.html#rfc.section.5.3) a duplicated header is semantically identical to concatenating them with a comma:

> A recipient MAY combine multiple field lines within a field section that have the same field name into one field line, without changing the semantics of the message, by appending each subsequent field line value to the initial field line value in order, separated by a comma (",") and optional whitespace ([OWS](https://httpwg.org/specs/rfc9110.html#whitespace), defined in [Section 5.6.3](https://httpwg.org/specs/rfc9110.html#whitespace)). For consistency, use comma SP.

> The order in which field lines with the same name are received is therefore significant to the interpretation of the field value; a proxy MUST NOT change the order of these field line values when forwarding a message.